plugins/logind/fu-plugin-logind.c

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <gio/gunixfdlist.h>
+#include <glib/gstdio.h>
+
+#include "fu-plugin-vfuncs.h"
+#include "fu-hash.h"
+
+struct FuPluginData {
+	GDBusProxy		*logind_proxy;
+	gint			 logind_fd;
+};
+
+void
+fu_plugin_init (FuPlugin *plugin)
+{
+	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
+	fu_plugin_alloc_data (plugin, sizeof (FuPluginData));
+}
+
+void
+fu_plugin_destroy (FuPlugin *plugin)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	if (data->logind_fd != 0)
+		g_close (data->logind_fd, NULL);
+	if (data->logind_proxy != NULL)
+		g_object_unref (data->logind_proxy);
+}
+
+gboolean
+fu_plugin_startup (FuPlugin *plugin, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	data->logind_proxy =
+		g_dbus_proxy_new_for_bus_sync (G_BUS_TYPE_SYSTEM,
+					       G_DBUS_PROXY_FLAGS_DO_NOT_CONNECT_SIGNALS |
+					       G_DBUS_PROXY_FLAGS_DO_NOT_LOAD_PROPERTIES,
+					       NULL,
+					       "org.freedesktop.login1",
+					       "/org/freedesktop/login1",
+					       "org.freedesktop.login1.Manager",
+					       NULL,
+					       error);
+	if (data->logind_proxy == NULL) {
```

Comment:
As part of startup do you want to do something similar to https://github.com/fwupd/fwupd/commit/6b8c2612251c8b19ec707d2fce13051e66bf73e2#diff-d70662bee1ff6feea3befa4f19a01b18 to make sure that it really is available?

Reply:
I guess it's a good idea. I don't think logind is ever autostarted, but it can't hurt.